test/Microsoft.ML.Functional.Tests/Common.cs

```diff
@@ -20,5 +20,34 @@ public static void CheckMetrics(RegressionMetrics metrics)
             Assert.True(metrics.L2 >= 0);
             Assert.True(metrics.RSquared <= 1);
         }
+
+        public static void AssertEqual(float[] array1, float[] array2)
+        {
+            Assert.NotNull(array1);
+            Assert.NotNull(array2);
+            Assert.Equal(array1.Length, array2.Length);
+
+            for (int i = 0; i < array1.Length; i++)
+                Assert.Equal(array1[i], array2[i]);
+        }
+
+        public static void AssertEqual(Schema schema1, Schema schema2)
+        {
+            Assert.NotNull(schema1);
+            Assert.NotNull(schema2);
```

Comment:
technically if they are both null they are equal...

Reply:
Technically yes. But internally in our library we consider nulls to not equal each other. :shruggy-emoticon:

---
In reply to: [256209640](https://github.com/dotnet/machinelearning/pull/2518#discussion_r256209640) [](ancestors = 256209640)